pkg/services/rpcsrv/server_test.go

```diff
@@ -3331,3 +3332,25 @@ func BenchmarkHandleIn(b *testing.B) {
 {"type": "Integer", "value": "42"}, {"type": "Boolean", "value": false}]]}`))
 	})
 }
+
+func TestFailedPreconditionShutdown(t *testing.T) {
+	_, srv, _ := initClearServerWithCustomConfig(t, func(c *config.Config) {
+		c.ApplicationConfiguration.RPC.Addresses = []string{"not an address"}
+	})
+
+	srv.Start()
+	require.Positive(t, len(srv.errChan)) // this is how Start reports internal failures
+
+	// TODO: #2626 use sync/atomic
+	var stopped atomic.Bool
+
+	go func() {
+		srv.Shutdown()
+		stopped.Store(true)
+	}()
+
+	// heuristic values, but in practice expected be legal
+	const waitInterval = 5 * time.Second
+	const checkInterval = 100 * time.Millisecond
+	require.Eventually(t, stopped.Load, waitInterval, checkInterval, "Shutdown should return")
```

Comment:
Polling intervals are almost always heuristics, we don't need these constansts, just use
```suggestion
        require.Eventually(t, stopped.Load, time.Second, 100 * time.Millisecond, "Shutdown should return")
```